### DISCUSSION TITLE
Read Description and Keyword Exif Tags

### DISCUSSION THREAD

**Discussion by mtnbiker777** [photoprism/photoprism#998] [General]
Coming at this from a different direction, is there a EXIF TAG I can write that would be read by Photoprism and displayed in the Description or Notes fields in the details tab for a photo?

**Comment by graciousgrey** [comment]
PhotoPrism reads the ImageDescription ExifTag

**Reply by mtnbiker777** [reply]
That is awesome, thank you. I tried it and it worked. I used EXIFTOOL to set it using 

exiftool -XPComment="White winged scoter"  -ImageDescription="White winged scoter" IMG_9830e.JPG

**Reply by mtnbiker777** [reply]
if I can push my luck, is there any tag I can add to a photo that would automatically create a label in Photoprism? If not I guess I's a sidecar file?

**Reply by lastzero** [reply]
Cat

**Reply by mtnbiker777** [reply]
I'm sorry lastzero, I'm not sure what you mean by Cat?

**Reply by lastzero** [reply]
Adding the keyword cat should add a photo to the label cat.

**Reply by mtnbiker777** [reply]
So I added the label Osprey for example to one of the pics, which creates the label. There are numerous pics with Osprey in the description, and they are found by searching for Osprey, I did a complete rescan of the library but there is still only one pic with the label Osprey. Am I missing something. So sorry if this is obvious, but I appreciate your responses, thank you.

**Comment by mtnbiker777** [comment]
so is there an EXIF tag that would end up properly populating the keyword and thus trigger the label? I've tried both ImageDescription and XPComment, both of which are being seen by Photoprism but are still not ending up in keyword, and thus not getting labelled.

**Reply by lastzero** [reply]
You're welcome to send related test files to hello@photoprism.app for debugging. It's possible that there is another keyword source or you've manually edited keywords, so that the keywords from the file get ignored or replaced.

**Reply by mtnbiker777** [reply]
Thanks for offering to help me, I've sent it over. Please let me know what I've done wrong :(

**Reply by lastzero** [reply]
Will do, but might take a couple of days.... we're super busy :/

**Reply by mtnbiker777** [reply]
Understood. Thank you for looking at it. I'm stumped.

**Comment by graciousgrey** [comment]
@mtnbiker777 ImageDescription is added as description and not as keyword, XPComment is not read at all as far as I know.

You can set keywords to your files using exiftool:

`exiftool -keywords+=banana 123.jpg`

If you like your keywords to be added as labels you need to do the following:
* For each label add it once manually to a photo
* Then trigger a complete rescan --> this will add all photos with the keyword matching the label to the label

**Reply by mtnbiker777** [reply]
Thank you for the reply. This helped a LOT. I've now got this working a lot better. I ran into one complication, I had spaces in my labels for example mute swan, which I resolved by adding -, so mute-swan. I have one remaining complication, labels seem to be case sensitive. Is there anyway to tell Photoprism to ignore case or do I need to insure my labels and my keywords on the photos are always in a particular case?

**Reply by graciousgrey** [reply]
Internally PhotoPrism saves keywords and labels as lower case. I guess the comparison runs on these lower case strings, so it should not be an issue if in your exif data the keywords are e.g. upper case. But to be sure test it with one example.

**Reply by dominikholler** [reply]
I just want to acknowledge that these two steps worked for me.

**Comment by graciousgrey** [comment]
An overview on what exiftool tags are read by Photoprism is now available: https://docs.photoprism.org/user-guide/advanced/metadata/
